src/rule_definition_tools.jl

```diff
@@ -1,4 +1,13 @@
 # These are some macros (and supporting functions) to make it easier to define rules.
+using Base.Meta
+
+macro strip_linenos(expr)
```

Comment:
Why is this a macro?

Reply:
It operates on the quote itself, rather than the interpolated expression.

Reply:
I see, can we add a comment to this effect?  or a docstring?

Reply:
Sure, feel free to add whatever you think would be helpful.

Reply:
Is the following correct?
```suggestion
"This should be put before a `quote`, to stop the quote from having line numbers recorded"
macro strip_linenos(expr)
```

Reply:
Well, that is how to use it, but it's a bit of an odd description of the macro. It just deleted linenumber nodes in whatever block you put it in front of. E.g. if you did:
```
@strip_linenos function foo(x)
<blah>
end
```
then that function wouldn't have line numbers either.